primitives/src/validator.rs

```diff
@@ -16,7 +16,25 @@ pub enum ValidatorError {
 
 #[derive(Deserialize, Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 #[serde(transparent)]
-pub struct ValidatorId(#[serde(with = "SerHex::<StrictPfx>")] [u8; 20]);
+pub struct ValidatorId(
+    #[serde(
+        deserialize_with = "validator_id_from_str",
+        serialize_with = "SerHex::<StrictPfx>::serialize"
+    )]
+    [u8; 20],
+);
+
+fn validator_id_from_str<'de, D>(deserializer: D) -> Result<[u8; 20], D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let validator_id = String::deserialize(deserializer)?;
+    if validator_id.is_empty() || validator_id.len() != 42 {
+        return Err(serde::de::Error::custom("invalid validator id".to_string()));
```

Comment:
This error is a bit vague, because on invalid hex string it is still an `invalid validator id`

Reply:
I am not sure what you mean by vague, it's a validator id param and if it's not a valid hex then it's an invalid validator id

Reply:
For debugging purposes it would be nice to distinguish between `empty/invalid length` and invalid `hex` string that you will get from `FromHex::from_hex`

Reply:
yeah i changed the error message to invalid length